src/main/java/com/google/sps/servlets/DataFromDatabase.java

```diff
@@ -0,0 +1,134 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/data-from-db")
+public class DataFromDatabase extends HttpServlet {
+
+  DatabaseClient dbClient;
+  private String[] selectedTables;
+
+  public void init() {
+    Spanner spanner = SpannerOptions.newBuilder().build().getService();
+    DatabaseId db = DatabaseId.of("play-user-data-beetle", "test-instance", "example-db");
+    this.dbClient = spanner.getDatabaseClient(db);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+
+    for (String table : selectedTables) {
```

Comment:
How does selectedTables get populated?

Reply:
From a post request (in the doPost below). When the form is submitted a post request happens, sending the selected values to the java servlet. I tried putting it in the get request instead, but the values were always null. Is there a better way around this?

Reply:
Fixed this - now gets populated in the doGet